packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,221 @@
+
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+}
+
+abstract class ProductBase extends Resource implements IProduct {
+  public abstract readonly productArn: string;
+  public abstract readonly productId: string;
+}
+
+/**
+ * Properties of provisioning artifacts
+ */
+export interface ProvisioningArtifactProperties {
+  /**
+   * The description of the provisioning artifact
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * Disable template validation.
+   * @default True
+   */
+  readonly disableTemplateValidation?: boolean;
+
+  /**
+   * The S3 template that points to the provisioning artifacts template
+   */
+  readonly template: Template;
+
+  /**
+   * The name of the provisioning artifact.
+   * @default - No provisioning artifact name provided
+   */
+  readonly name?: string;
+}
+
+/**
+ * Properties for a Cloudformation Product
+ */
+export interface CloudFormationProductProps {
+  /**
+   * The owner of the product.
+   */
+  readonly owner: string;
+
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+
+  /**
+   * The configuration of the provisioning artifact (also known as a version).
+   */
+  readonly provisioningArtifacts: ProvisioningArtifactProperties[];
+
+  /**
+   * The language code.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * The description of the product.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * The distributor of the product.
+   * @default - No distributor provided
+   */
+  readonly distributor?: string;
+
+  /**
+   * Whether to give provisioning artifacts a new unique identifier when the product attributes or provisioning artifacts is updated
+   * @default false
+   */
+  readonly replaceProvisioningParameters?: boolean;
+
+  /**
+   * The support information about the product
+   * @default - No support description provided
+   */
+  readonly supportDescription?: string;
+
+  /**
+   * The contact email for product support.
+   * @default - No support email provided
+   */
+  readonly supportEmail?: string;
+
+  /**
+   * The contact URL for product support.
+   * @default - No support URL provided
+   */
+  readonly supportUrl?: string;
+}
+
+/**
+ * Abstract class for Service Catalog Product.
+ */
+export abstract class Product extends ProductBase {
+  /**
+   * Creates a Product construct that represents an external product.
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs Product import properties
+   */
+  public static fromProductArn(scope: Construct, id: string, attrs: ProductAttributes): IProduct {
+    const parts = Stack.of(scope).parseArn(attrs.productArn);
+    if (!parts.resourceName) {
+      throw new Error(`Product arn ${attrs.productArn} missing Product ID during import from attributes`);
```

Comment:
I feel like the wording in this message is a little awkward... can we use [this one](https://github.com/aws/aws-cdk/blob/e47a01feaab8579a8f120b44d26871b8a586df78/packages/%40aws-cdk/aws-servicecatalog/lib/portfolio.ts#L163-L163) instead?